misk/web/@misk/components/src/components/TopbarComponentDimensionAware.tsx

```diff
@@ -0,0 +1,388 @@
+  font-size: 24px !important;
+  text-decoration: none;
+  text-transform: uppercase;
+  letter-spacing: 0px;
+  margin-right: 30px;
+  color: ${color.platinum};
+  min-width: fit-content;
+
+  a {
+    color: ${color.platinum} !important;
+    letter-spacing: 1px;
+    text-decoration: none;
+    &:hover {
+      color: ${color.white} !important;
+      text-decoration: none;
+    }
+  }
+`
+
+const MiskNavbarHeadingEnvironment = styled(MiskNavbarHeading)`
+  color: ${props => props.color} !important;
+  min-width: 0;
+`
+
+const MiskLink = styled(Link)`
+  color: ${color.gray};
+  text-decoration: none;
+  &:hover {
+    color: ${color.white};
+    text-decoration: none;
+  }
+`
+
+const MiskLinkHome = styled(MiskLink)`
+  min-width: fit-content;
+`
+
+const MiskNavbarButton = styled(Button)`
+  background-color: ${color.cadet} !important;
+  box-shadow: none !important;
+  background-image: none !important;
+  top: 15px;
+  left: 15px;
+  position: absolute;
+  z-index: 1020;
+`
+
+const MiskNavbarIcon = styled(Icon)`
+  color: ${color.gray} !important;
+  &:hover {
+    color: ${color.white};
+  }
+`
+
+const MiskCollapse = styled(Collapse)`
+  color: ${color.white};
+  background-color: ${color.cadet};
+  display: block;
+  margin: 60px -20px 0 -20px;
+`
+
+const MiskMenu = styled.div`
+  min-height: 250px;
+  padding: 50px 0px;
+  @media (max-width: 768px) {
+    padding: 50px 20px;
+  }
+  overflow-y: scroll;
+  max-height: 100vh;
+`
+
+const MiskMenuNavbarItems = styled.div`
+  display: inline-block;
+`
+
+const MiskMenuLinks = styled(FlexContainer)`
+  padding-bottom: 35px;
+`
+
+const MiskMenuLink = styled(MiskLink)`
+  font-size: 16px;
+  flex-basis: 300px;
+  padding: 5px 0;
+  color: ${color.platinum};
+`
+
+const MiskMenuCategory = styled.span`
+  font-size: 24px;
+  color: ${color.gray};
+  letter-spacing: 0px;
+  display: block;
+`
+
+const MiskMenuDivider = styled.hr`
+  border-color: ${color.gray};
+  margin: 5px 0 10px 0;
+`
+
+const MiskBanner = styled.span`
+  background-color: ${props => props.color} !important;
+  color: ${color.accent} !important;
+  text-align: center;
+  font-weight: 600;
+  padding: 5px 10px;
+  position: fixed !important;
+  width: 100%;
+  top: 70px;
+  left: 0px;
+  z-index: 1010 !important;
+
+  a {
+    font-weight: 300;
+    color: ${color.accent};
+    text-decoration: underline;
+    letter-spacing: 1px;
+    &:hover {
+      color: ${color.white};
+      text-decoration: underline;
+    }
+  }
+`
+
+const environmentToColor = (environment: Environment) => {
+  switch (environment) {
+    case Environment.DEVELOPMENT:
+      return color.blue
+    case Environment.TESTING:
+      return color.indigo
+    case Environment.STAGING:
+      return color.green
+    case Environment.PRODUCTION:
+      return color.red
+  }
+}
+
+export class TopbarComponentDimensionAware extends React.Component<
+  IDimensionAwareProps & ITopbarProps,
+  {}
+> {
+  public state = {
```

Comment:
This is probably fine, and this mostly comes from my tendency to avoid state whenever possible, but a nice pattern can be to use a HOC to pass in the state and its handlers to the component. Totally unnecessary, but I like the separation of concerns you get.

Reply:
Yeah I've tried to minimize any local state to simple UI methods. I've further reduced this massive component into smaller ones that should help limit the scope that the State is needed. HOC is definitely a good pattern.